gbus/worker.go

```diff
@@ -219,48 +223,64 @@ func (worker *worker) invokeDeadletterHandler(delivery amqp.Delivery) {
 }
 
 func (worker *worker) processMessage(delivery amqp.Delivery, isRPCreply bool) {
+	var span opentracing.Span
+	var ctx context.Context
+	var spanOptions []opentracing.StartSpanOption
+
+	spCtx, err := amqptracer.Extract(delivery.Headers)
+	if err != nil {
+		worker.log().WithError(err).Error("could not extract SpanContext from headers")
+	} else {
+		spanOptions = append(spanOptions, opentracing.FollowsFrom(spCtx))
+	}
+	span, ctx = opentracing.StartSpanFromContext(context.Background(), "processMessage", spanOptions...)
 
 	//catch all error handling so goroutine will not crash
 	defer func() {
 		if r := recover(); r != nil {
 			logEntry := worker.log().WithField("worker", worker.consumerTag)
 			if err, ok := r.(error); ok {
+				span.LogFields(slog.Error(err))
 				logEntry = logEntry.WithError(err)
 			} else {
 				logEntry = logEntry.WithField("panic", r)
 			}
-
+			span.LogFields(slog.String("panic", "failed to process message"))
 			logEntry.Error("failed to process message")
 		}
+		span.Finish()
 	}()
 
 	worker.log().WithFields(log.Fields{"worker": worker.consumerTag, "message_id": delivery.MessageId}).Info("GOT MSG")
-	// worker.log("%v GOT MSG - Worker %v - MessageId %v", worker.svcName, worker.consumerTag, delivery.MessageId)
-	/*TODO:FIX Opentracing
-	spCtx, _ := amqptracer.Extract(delivery.Headers)
-	sp := opentracing.StartSpan(
-		"processMessage",
-		opentracing.FollowsFrom(spCtx),
-	)
-	if sp != nil {
-		defer sp.Finish()
-	}
-	*/
-	// Update the context with the span for the subsequent reference.
 
 	//handle a message that originated from a deadletter exchange
 	if worker.isDead(delivery) {
+		span.LogFields(slog.Error(errors.New("delivery is dead")))
```

Comment:
"delivery is dead" is a bit misleading, think a better trace entry would be "handling deadletter delivery"  or something like that.